arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -97,95 +111,251 @@ static void _region_init(u32_t index, u32_t region_base,
 		    (u32_t)SYSMPU->WORD[index][1],
 		    (u32_t)SYSMPU->WORD[index][2],
 		    (u32_t)SYSMPU->WORD[index][3]);
+
 }
 
-/**
- * This internal function is utilized by the MPU driver to parse the intent
- * type (i.e. THREAD_STACK_REGION) and return the correct region index.
- */
-static inline u32_t _get_region_index_by_type(u32_t type)
+static int _region_allocate_and_init(const u8_t index,
+	const struct nxp_mpu_region *region_conf)
 {
-	u32_t region_index;
-
-	__ASSERT(type < THREAD_MPU_REGION_LAST,
-		 "unsupported region type");
+	/* Attempt to allocate new region index. */
+	if (index > (_get_num_regions() - 1)) {
 
+		/* No available MPU region index. */
+		return -EINVAL;
```

Comment:
Add a warning message as in arm_mpu.c
LOG_ERR("Failed to allocate new MPU region %u\n", index);

Reply:
Thanks. Will do.